pkgs/development/python-modules/dendropy/default.nix

```diff
@@ -1,45 +1,47 @@
 {
   lib,
+  stdenv,
+  checkMeta,
   buildPythonPackage,
   fetchFromGitHub,
   pytestCheckHook,
   pythonOlder,
+  nix-update-script,
+  setuptools,
+  paup,
+  paupIntegration ? (checkMeta.checkValidity paup).valid == "yes"
 }:
 
 buildPythonPackage rec {
   pname = "dendropy";
-  version = "4.5.1";
-  format = "setuptools";
+  version = "5.0.1";
 
-  disabled = pythonOlder "3.7";
+  pyproject = true;
+  build-system = [ setuptools ];
 
   src = fetchFromGitHub {
     owner = "jeetsukumaran";
     repo = pname;
-    rev = "v${version}";
-    hash = "sha256-FP0+fJkkFtSysPxoHXjyMgF8pPin7aRyzmHe9bH8LlM=";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-4VkFkY4/hDfb6elOSTDia6tRJm73c1YP9nBW2dJsxsI=";
   };
 
-  nativeCheckInputs = [ pytestCheckHook ];
+  postPatch = ''
+    substituteInPlace setup.py --replace '["pytest-runner"],' '[],'
+  '';
 
-  disabledTests = [
-    # FileNotFoundError: [Errno 2] No such file or directory: 'paup'
-    "test_basic_split_count_with_incorrect_rootings_raises_error"
-    "test_basic_split_count_with_incorrect_weight_treatment_raises_error"
-    "test_basic_split_counting_under_different_rootings"
-    "test_group1"
-    # AssertionError: 6 != 5
-    "test_by_num_lineages"
-    # AttributeError: module 'collections' has no attribute 'Iterable'
-    "test_findall_multiple"
-  ];
+  nativeCheckInputs = [ pytestCheckHook ];
 
   pythonImportsCheck = [ "dendropy" ];
 
-  meta = with lib; {
+  env.DENDROPY_PAUP_EXECUTABLE_PATH = if paupIntegration then lib.getExe paup else "NONE";
```

Comment:
It's a bit tricky.
It seems reasonable to leave it to upstream to make sure the paup integration works, and we disable it.

Reply:
Annoyingly, upstream *doesn't* make sure the paup integration works. If you remove this line, you'll get failed tests because it can't find the paup binary.

Reply:
In any case, paup is marked as unfree, so our CI will not run the test either.

Reply:
> In any case, paup is marked as unfree, so our CI will not run the test either.

There are four tests that test PAUP integration. They are disabled when `DENDROPY_PAUP_EXECUTABLE_PATH` is set to `NONE`. They are *not* disabled when the `DENDROPY_PAUP_EXECUTABLE_PATH` environment variable is not set. Then, dendropy just assumes that PAUP is in the `PATH`.

Reply:
The upstream looks like it is running an integration test.
https://github.com/jeetsukumaran/DendroPy/blob/a27a6d46e5a7e1c9573b3022a18f3005cd8b996e/.github/workflows/ci.yaml#L30-L37

I'm not willing to expose `checkValidity` just for the four tests in this package.

Reply:
You're right in regards of just four unit tests being insufficient to expose `checkValidity`. I meant to additionally figure out a way to make PAUP* integration work for dependents of dendropy.

Also, there are a couple other packages where access to `checkValidity` is useful (https://github.com/NixOS/nixpkgs/pull/310138's integration with monado, for example). In fact, any open source project with an optional non-free dependency could really make use of `checkValidity`.

Reply:
I'm not very familiar with check-meta, so I can't decide if it should be exposed.
However, I agree that there are a number of useful examples.
How about implementing this suggestion in another PR?
I think it would get more people involved in the discussion.

Reply:
Okay, then I will make it a separate PR.